### PR TITLE
 rocmPackages.miopen: fix bnorm and dropout kernels

### LINKED ISSUE #498893: python313Packages.torchWithRocm / rocmPackages.miopen: ROCm training fails with MIOpen SQLite schema mismatch ("table config has no column named mode")
### Nixpkgs version

- Unstable (26.05)

### Describe the bug

PyTorch training on ROCm fails when MIOpen is enabled. The failure happens inside MIOpen’s packaged performance/tuning DB handling, with this SQLite error:

SQLite prepare error: Internal error while accessing SQLite database: table config has no column named mode

### Steps to reproduce

  1. Use a NixOS system with an AMD Radeon RX 9070 XT (gfx1201).
  2. Use a nixpkgs environment with ROCm-enabled PyTorch (python3Packages.torchWithRocm) and MIOpen.
  3. Run a normal object detection training job through Ultralytics / PyTorch on ROCm.
  4. Let training begin normally until the first real GPU training step reaches MIOpen-backed ops such as convolution or batch norm.
  5. Observe that training fails inside MIOpen with SQLite schema errors.

  A simplified reproduction from my project looked like this:

  1. Build a Python environment containing:
      - python3Packages.torchWithRocm
      - python3Packages.torchvision
      - python3Packages.ultralytics
  2. Run YOLO training on ROCm with device=auto.
  3. Observe failure in MIOpen with:
      - table config has no column named mode
      - No solver found
      - earlier related MIOpen DB/FindDb failures depending on env config

### Expected behaviour

Training completes with ROCm and MIOpen enabled.

### Screenshots

_No response_

### Relevant log output

```console
MIOpen(HIP): Error [FindSolutionImpl] Search Failed: 2x16x16x512x512x1x1x1x0xNCHWxTrnxFP32x1BnFwdTrainingSpatial: nixos:/build/source/src/sqlite_db.cpp:298: SQLite prepare error: Internal error while accessing SQLite database: table config has no column named mode
MIOpen(HIP): Error [SearchForSolutions] BnFwdTrainingSpatial: Applicable Solver not succeeded.
MIOpen Error: nixos:/build/source/src/include/miopen/find_solution.hpp:482: No solver found.
miopenStatusNotImplemented
```

### Additional context

  - GPU: AMD Radeon RX 9070 XT (gfx1201)
  - PyTorch version: 2.10.0
  - Ultralytics version: 8.4.16
  - ROCm version (MIOpen version): 7.2.0
  - Training only worked after explicitly disabling MIOpen in PyTorch:
      - torch.backends.cudnn.enabled = False on ROCm
      - However, it's much slower.
  - I also hit earlier MIOpen issues around HIPRTC architecture propagation and cache/DB handling, but the persistent blocker was the table config has no column named mode schema mismatch.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.16, NixOS, 26.05 (Yarara), 26.05.20260306.aca4d95`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.33.3`
 - channels(root): `"nixos-24.11"`
 - nixpkgs: `/nix/store/3x925rhb3gjikmqkyqigjcdfqm1hsn6a-source`

### Notify maintainers

@mschwaig @Flakebi @GZGavinZhao @LunNova

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by LunNova**
Fixes #498893

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

### CODE REVIEW

**Comment by justinkb** [comment]
Cool, I'll look through the miopen code to see if there's any other headers it might be including for runtime compilation in other scenarios

**Review by Flakebi — APPROVED** [review]
Thanks!

**Review by krishnans2006 — APPROVED** [review]

**Comment by LunNova** [comment]
Landing this one now, can handle anything else missed in a followup.
